From fd272f86829c41a32c06fa8440eae5a4bfd7483e Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Sat, 21 Mar 2020 12:46:15 +0100
Subject: [PATCH] Use quasi standard BUILD_QCH option

As found in use consistently in KDE Frameworks and some other libraries.
---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 588f2e000..886916c5a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,6 +29,8 @@ option(BUILD_KF5 "Build for KDE Frameworks 5" ON)
 
 option(BUILD_PHP_IMPORT "Build with php import support" ON)
 
+option(BUILD_QCH "Build API documentation in QCH format (for e.g. Qt Assistant, Qt Creator & KDevelop)" OFF)
+
 # where to look first for cmake modules, before ${CMAKE_ROOT}/Modules/ is checked
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
 include(Macros)
@@ -236,7 +238,9 @@ add_unstable_feature(OBJECT_DIAGRAM) # see https://bugs.kde.org/show_bug.cgi?id=
 find_package(Doxygen OPTIONAL_COMPONENTS dot)
 if(DOXYGEN_FOUND)
     message(STATUS "found doxygen package")
-    find_program(QHELPGENERATOR_EXECUTABLE qhelpgenerator)
+    if (BUILD_QCH)
+        find_program(QHELPGENERATOR_EXECUTABLE qhelpgenerator)
+    endif()
 
     set(DOXYGEN_ABBREVIATE_BRIEF "The $name class" "The $name widget")
     set(DOXYGEN_CALL_GRAPH YES)
-- 
2.25.2

